pytensor/tensor/basic.py

```diff
@@ -3865,7 +3776,14 @@ def alloc_diag(diag, offset=0, axis1=0, axis2=1):
         axes = axes[:axis2] + [last_idx + 2] + axes[axis2:]
         result = result.transpose(axes)
 
-    return result
+    return AllocDiag(
+        inputs=[diag],
+        outputs=[result],
+        offset=offset,
+        axis1=axis1,
+        axis2=axis2,
+        inline=True,
```

Comment:
If you inline it's going to disappear immediately. This happens first in the rewrite schedule. So you don't want to do it.

We may want to inline later but we have to think about it. May need two phases